src/test/java/com/mappy/fpm/batches/tomtom/shapefiles/RoadShapefileTest.java

```diff
@@ -29,8 +30,12 @@ public static void loadPbf() {
 
         TomtomFolder tomtomFolder = mock(TomtomFolder.class);
         when(tomtomFolder.getFile("nw.shp")).thenReturn("src/test/resources/tomtom/road/nw.shp");
+        when(tomtomFolder.getZone()).thenReturn("mockzone");
+        when(tomtomFolder.getInputFolder()).thenReturn("/this/is/a/path/mockinputfolder");
 
-        when(roadTagger.tag(any(Feature.class))).thenReturn(of("ref:tomtom", "12500001097987"));
+        Map<String, String> mockedMap = newHashMap();
```

Comment:
What would be the benefit of such a change ?

Reply:
`RoadShapefile` now modifies the tags returned by `RoadTagger` to add country/zone, thus an `ImmutableMap` wasn't usable anymore.

Reply:
Letting `RoadTagger` set the tags instead of `RoadShapefile` would let the tagging stay at one place.

Reply:
It seems a bit awkward to add some feature-independant tagging in `RoadTagger`. Moreover, country and zone aren't even specific to "road" elements.

Why would it be important to keep all the taggings in `RoadTagger` ?

The way I see it, if one wanted to do things a bit more cleanly, one would create a general purpose tagger (or maybe `SourceTagger` ?) that adds its tags over the road-specific ones, but this seems overkill at the moment.

Reply:
A single entry point for tagging a road would seem cleaner to me, as one may have some other use cases than the one we currently have (e.g. one may want to use another file format for the output), and it would be easier to search for the tagging behaviour knowing that the job is traced to a single entry point.

I see your point though about data vs. metadata.